File: proxy/models/movie.py
from typing import Optional, Dict, List
from dataclasses import dataclass
from proxy.models.base import Images, Platform, Author


@dataclass
class Movie:
    id: int
    title: str
    original_title: str
    content_type: str
    description: Optional[str] = None
    image_url: Optional[str] = None
    tagline: Optional[str] = None
    imdb_id: Optional[str] = None
    release_date: Optional[str] = None
    duration_minutes: Optional[int] = None
    status: Optional[str] = None
    authors: Optional[List[Author]] = None
    images: Optional[Images] = None
    platforms: Optional[Dict[str, List[Platform]]] = None

    def to_dict(self) -> Dict:
        result = {
            'id': self.id,
            'type': self.content_type,
            'imdb_id': self.imdb_id,
            'title': self.title,
            'original_title': self.original_title,
            'tagline': self.tagline,
            'description': self.description,
            'image_url': self.image_url,
            'release_date': self.release_date,
            'status': self.status,
            'duration_minutes': self.duration_minutes,
        }

        if self.authors:
            result['authors'] = [author.to_dict() for author in self.authors]

        if self.images:
            result['images'] = self.images.to_dict()

        if self.platforms:
            result['platforms'] = {
                country: [platform.to_dict() for platform in platforms]
                for country, platforms in self.platforms.items()
            }

        return result

File: internal/models/movie.go
package models

type Movie struct {
	ID              string                `json:"id"`
	Title           string                `json:"title"`
	OriginalTitle   string                `json:"original_title"`
	ContentType     string                `json:"type"`
	Description     *string               `json:"description,omitempty"`
	ImageURL        *string               `json:"image_url,omitempty"`
	Tagline         *string               `json:"tagline,omitempty"`
	ImdbID          *string               `json:"imdb_id,omitempty"`
	ReleaseDate     *string               `json:"release_date,omitempty"`
	DurationMinutes *int                  `json:"duration_minutes,omitempty"`
	Status          *string               `json:"status,omitempty"`
	Authors         []Author              `json:"authors,omitempty"`
	Images          *Images               `json:"-"`
	Platforms       map[string][]Platform `json:"-"`
}

type MovieResponse struct {
	ID              string                `json:"id"`
	Type            string                `json:"type"`
	ImdbID          *string               `json:"imdb_id,omitempty"`
	Title           string                `json:"title"`
	OriginalTitle   string                `json:"original_title"`
	Tagline         *string               `json:"tagline,omitempty"`
	Description     *string               `json:"description,omitempty"`
	ImageURL        *string               `json:"image_url,omitempty"`
	ReleaseDate     *string               `json:"release_date,omitempty"`
	Status          *string               `json:"status,omitempty"`
	DurationMinutes *int                  `json:"duration_minutes,omitempty"`
	Authors         []Author              `json:"authors,omitempty"`
	Images          []ImageEntry          `json:"images,omitempty"`
	Platforms       map[string][]Platform `json:"platforms,omitempty"`
}

func (m *Movie) ToResponse() MovieResponse {
	resp := MovieResponse{
		ID:              m.ID,
		Type:            m.ContentType,
		ImdbID:          m.ImdbID,
		Title:           m.Title,
		OriginalTitle:   m.OriginalTitle,
		Tagline:         m.Tagline,
		Description:     m.Description,
		ImageURL:        m.ImageURL,
		ReleaseDate:     m.ReleaseDate,
		Status:          m.Status,
		DurationMinutes: m.DurationMinutes,
		Authors:         m.Authors,
	}

	if m.Images != nil {
		resp.Images = m.Images.ToList(DefaultImagesSize)
	}

	if m.Platforms != nil {
		resp.Platforms = m.Platforms
	}

	return resp
}
